### PR TITLE
Fix rt module and duplicate tickets

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: inspirehep/modules/rt/api.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from __future__ import absolute_import, division, print_function
+
+import backoff
+import rt
+
+from .tickets import create_ticket_with_template
+
+
+@backoff.on_exception(backoff.expo, rt.ConnectionError, base=4, max_tries=5)
+def submit_ticket(obj, queue, template, context, requestors, recid,
```
Comment (puntonim) [inline inspirehep/modules/rt/api.py]: Why did you move `submit_ticket()` to a new `api.py` module? What makes this different from all the other functions in the new `tasks.py` module?

Comment (drjova) [reply inspirehep/modules/rt/api.py]: the ``task`` module contains ``wraps`` which are used in the workflow

File: inspirehep/modules/rt/api.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""INSPIRE-HEP ticketing system API."""
+
+from __future__ import absolute_import, division, print_function
+
+import backoff
+import rt
```
Comment (puntonim) [inline inspirehep/modules/rt/api.py]: `rt` is an external service and you renamed this module `rt`. Thus now an import like `import rt` is very confusing (our module or the external service?).

**Comment by david-caro** [comment]
This PR makes some design decisions that I think should be discussed, can we discuss it at some point before merging this? (it's not blocking anything right?)
We might want to discuss also what types of changes should be agreed or not, to what point can we decide stuff by ourselves or when should we bring in the rest of the team (retrospective material maybe).

File: inspirehep/modules/rt/tasks.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
```
Comment (puntonim) [inline inspirehep/modules/rt/tasks.py]: To me, all workflow tasks should go all together in a workflow package (or layer).
And clients of external services (like `InspireRtExt`) should go to a service layer and have a similar (or even shared) interface.

And in general these kind of refactorings are actual architectural decisions that should be discussed in a refinement. Only with the input of everyone we can take the best decision and share the knowledge.

**Comment by drjova** [comment]
I had a PR for moving rt outside of ``inspirehep.utils`` for two weeks now, I just moved the code so we can easily discuss were we'll move it i.e. in a separated module or so. Is not a design change it is just a way to make our life easier and help us separate it in the future

**Comment by michamos** [comment]
> This PR makes some design decisions that I think should be discussed, can we discuss it at some point before merging this? (it's not blocking anything right?)

It's supposed to fix the duplicate ticket issue. If you want to take your time and discuss what to do with code architecture, can you at least split off and merge the fix for that issue?
